src/Dict.h

```diff
@@ -639,6 +650,17 @@ template <typename T> class Dictionary
 				max_entries = num_entries;
 			if ( num_entries > ThresholdEntries() )
 				SizeUp();
+			// if space_distance is too great, performance decreases. we need to sizeup for
+			// performance.
+			else if ( space_distance_samples > detail::MIN_SPACE_DISTANCE_SAMPLES && 
+				space_distance_sum > space_distance_samples * detail::SPACE_DISTANCE_THRESHOLD &&
+			          int(num_entries) > detail::MIN_DICT_LOAD_FACTOR_100 * Capacity() / 100 )
+				{
+				SizeUp();
+				// reset performance metrics.
+				space_distance_sum = 0;
```

Comment:
Should these distance metrics be invalidated after the other `SizeUp()` calls as well?

Mostly wondering about corner cases where SizeUp() happens due to ThresholdEntries() (e.g. line 651) and then another one due to the distance metrics which aren't valid for the new size of the dictionary anymore. Also generally seems more consistent to reset these metrics unconditionally after resizing (maybe even reset in SizeUp() ? ).

Reply:
Good idea. To avoid the consecutive, probably unnecessary SizeUp().